oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/ClaimAccessor.java

```diff
@@ -98,8 +101,24 @@ default Instant getClaimAsInstant(String claim) {
 		if (Instant.class.isAssignableFrom(claimValue.getClass())) {
 			return (Instant) claimValue;
 		}
+		if (String.class.isAssignableFrom(claimValue.getClass())) {
+			try {
+				return Instant.from(DateTimeFormatter.ISO_INSTANT.parse((String)claimValue));
```

Comment:
The second one: DateTimeFormatter.ISO_ZONED_DATE_TIME works for UTC timestamps too (i.e. ending with Z), so I'm not sure if doing the ISO_INSTANT first is necessary, but I'll keep it as per your request.

Reply:
ok let's remove `DateTimeFormatter.ISO_INSTANT` and let `DateTimeFormatter.ISO_ZONED_DATE_TIME` handle both cases. Can you ensure we have 2 tests for:
- 2011-12-03T10:15:30Z
- 2011-12-03T10:15:30+01:00